napari/layers/image/_tests/test_image.py

```diff
@@ -646,3 +647,17 @@ def test_data_to_world_2d_scale_translate_affine_composed():
         image._data_to_world.affine_matrix,
         ((12, 0, -16), (0, 3, 12), (0, 0, 1)),
     )
+
+
+@pytest.mark.parametrize('scale', ((1, 1), (-1, 1), (1, -1), (-1, -1)))
+@pytest.mark.parametrize('angle_degrees', range(-180, 180, 30))
+def test_rotate_with_reflections_in_scale(scale, angle_degrees):
+    # See the GitHub issue for more details:
+    # https://github.com/napari/napari/issues/2984
+    data = np.ones((4, 3))
+    rotate = coerce_rotate(angle_degrees, 2)
```

Comment:
Sure. I think I learned `coerce_*` from somewhere else in the code (possibly related to Pydantic's validators), but agreed that it's not a great name and the positional `ndim` is more clear as a keyword argument, where the default ndim can be defined by the shape of the array-like input.

I'll probably go with `*_to_matrix` name you suggest or maybe more generally `*_to_array` to indicate that the output is a numpy array. The `ndim` argument is really just about padding the arrays rather than broadcasting to new array dimensions, so I'll probably avoid `broadcast_*`.